src/virtual_dom/patch.rs

```diff
@@ -1,78 +1,250 @@
 //! This module contains code related to patching the VDOM. It can be considered
 //! a subset of the `vdom` module.
 
-use super::{El, IntoNodes, Mailbox, Node};
+use super::{El, IntoNodes, Mailbox, Node, Text};
 use crate::app::App;
 use crate::browser::dom::virtual_dom_bridge;
 use wasm_bindgen::JsCast;
 use web_sys::Document;
 
+mod patch_gen;
+
+// We assume that when we run this, the new vdom doesn't have assigned `web_sys::Node`s -
+// assign them here when we create them.
+
+#[inline]
```

Comment:
In this case function bodies are small and they are used once or twice.
I use the `inline` in cases where  copy-paste of the function body is OK:
* function body is small;
* function called form few places.

In this particular case the binary sizes is:
With inline: 2398969 bytes
Without inline: 2398953 bytes
12 bytes is not so much.

Reply:
Is it in a debug mode? What about release mode + configured wasm-opt for size?

`Cargo.toml`
```toml
[profile.release]
lto = true
opt-level = 'z'
codegen-units = 1

[package.metadata.wasm-pack.profile.release]
wasm-opt = ['-Os']
```

Can we find out if those optimizations don't replace our inlines with their original function calls (or optimize them away by another way)? And what do you think about speed boost?

I'm asking because on the first glance they look like a premature optimization (or no optimization at all if the size is a trade-off and speed boost isn't reasonable big). And it's another thing we have to think about while writing and reading code, doing code reviews and test (once the bechmarks are configured).

Reply:
I have `inlines` on the "future optimizations list" - https://github.com/seed-rs/seed/issues/385. So I would like to know your opinion - if the trade-offs are good for our users and if it's the right time to do them.

Reply:
el_key example built in release mode with -Os produces exactly same binary (crc32 is same) with and without `#[inline]` attributes. Seems like size optimization comicality disables inlining.

Reply:
It was wrong file.
402787  pkg/el_key_bg_opt_iline.wasm
404284  pkg/el_key_bg_opt_noiline.wasm

Reply:
So... let's assume we add `inlines` for all similar function - let's say there would be 100 inlines total (there are 13 only in `patch.rs` and I've removed many inlines from the original fetch library so I assume the number can be also bigger with new Seed services (like `web_socket`)).
(404284 - 402787) / 13 * 100 = **11515** ; If the user compiles his project for speed, the size can be 1-2x as big => **23 030** (just theoretical value).
I'm not really sure if it's a good trade-off because we don't have problems with speed but with the file size. And instructions / rules for contributors and marks in the code would be just additional annoyance.
I think I don't want it in the code unless we find out that there is a big improvement that makes Seed faster than e.g. React in benchmarks. Then it will make sense for the marketing purposes; I assume users don't notice any differences in their apps.

Reply:
I removed all inline attributes. We can try them later when proper benchmarks are available.